### PR TITLE
Dispose widgets to stop leaks.

### PR DESCRIPTION
**Pull request by polina-c**
The leaks are detected by [leak-tracker](https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/memory/panes/leaks/LEAK_TRACKING.md) for money app.

We will test cover the change after the leak tracker release.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_position.dart [hunk 1]
```diff
@@ -976,6 +976,7 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
   void dispose() {
     activity?.dispose(); // it will be null if it got absorbed by another ScrollPosition
     _activity = null;
+    isScrollingNotifier.dispose();
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/scroll_position.dart:979]: Great to see the leak detection tool catching issues like this!

**Review by goderbauer — APPROVED** [review]
Nice!

LGTM

File: packages/flutter/lib/src/material/ink_well.dart [hunk 2]
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// We disable these lints, because they relate to InkResponseState
+// that is actually private, but visible for testing.
+// ignore_for_file: public_member_api_docs, library_private_types_in_public_api
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/ink_well.dart:7]: Please don't do this. We should try to keep that object private. 

If we can't keep it private, we will have to add docs. But We should really try hard to not make this public.

**Review by goderbauer — CHANGES_REQUESTED** [review]

File: packages/flutter/lib/src/material/ink_well.dart (hunk 2 above)
Comment (polina-c) [reply packages/flutter/lib/src/material/ink_well.dart]: Agreed.
We normally do not test disposal as the testing is complicated. So, I rolled back testing and will request the testing exempt. When we release the leak detector, we will be able to test cover such things.

**Comment by polina-c** [comment]
Requested exemption: https://discord.com/channels/608014603317936148/608018585025118217/1001186592179896450

**Comment by Hixie** [comment]
test-exempt: will be tested by https://github.com/flutter/flutter/issues/108322

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by dnfield — APPROVED** [review]

**Pull request merged**